src/datasets/table.py

```diff
@@ -145,31 +145,30 @@ def __deepcopy__(self, memo: dict):
         return _deepcopy(self, memo)
 
     def __getstate__(self):
-        state = self.__dict__.copy()
         # We can't pickle objects that are bigger than 4GiB, or it causes OverflowError
         # So we write the table on disk instead
         if self.table.nbytes >= config.MAX_TABLE_NBYTES_FOR_PICKLING:
-            table = state.pop("table")
+            table = self.table
             with tempfile.NamedTemporaryFile("wb", delete=False, suffix=".arrow") as tmp_file:
                 filename = tmp_file.name
                 logger.debug(
                     f"Attempting to pickle a table bigger than 4GiB. Writing it on the disk instead at {filename}"
                 )
                 _write_table_to_file(table=table, filename=filename)
-                state["path"] = filename
-                return state
+                return {"path": filename}
```

Comment:
Are there other items (previously in state) that we are missing now? Or just with `path` is enough?

Reply:
We only need "path" in this case:

- There were other items yes: _batches, _schema and _offsets. However we don't want to pickle _batches or it causes memory issues. Therefore all these attributes are reloaded when Table.__init__ is called.

- Then the only thing left to pickle is `table`. But if it's bigger than 4GB we are writing it on disk since pickle doesn't support pickling objects bigger than 4GB by default. In this case we just pickle the path to the table on disk